### PR TITLE
CMake: Require sqlite3 only on Unix

### PR DESCRIPTION
**Pull request by FlexW**
Signed-off-by: Felix Weilbach <felix.weilbach@nextcloud.com>

### CODE REVIEW

File: src/csync/CMakeLists.txt [hunk 1]
```diff
@@ -19,7 +19,15 @@ include(DefineOptions.cmake)
 
 include(DefineInstallationPaths)
 
-find_package(SQLite3 3.8.0 REQUIRED)
+if (APPLE OR WIN32)
```
Comment (mgallien) [inline src/csync/CMakeLists.txt:22]: Please keep only there the automatic choice for using the embedded SQLite copy here and then fill in all variables later.
So
```
if (APPLE OR WIN32)
    set(USE_OUR_OWN_SQLITE3 TRUE)
else()
    find_package(SQLite3 3.8.0 REQUIRED)
endif()
```

and later if `USE_OUR_OWN_SQLITE3` is true, set the needed variables

**Comment by nicolasfella** [comment]
I'm not sure about this..

Wouldn't it make sense to have all of the special-casing inside FindSQLite3.cmake and have it define an exported target?

csync would only have something like `target_link_libraries("${csync_NAME}" PUBLIC SQLite3::SQLite3)` and all of the complexity would be hidden away.

File: src/csync/CMakeLists.txt (hunk 1 above)
Comment (FlexW) [reply src/csync/CMakeLists.txt:22]: I changed it and introduced and made sqlite it's own target, to get rid of the extra if branch.

**Comment by FlexW** [comment]
> I'm not sure about this..
> 
> Wouldn't it make sense to have all of the special-casing inside FindSQLite3.cmake and have it define an exported target?
> 
> csync would only have something like `target_link_libraries("${csync_NAME}" PUBLIC SQLite3::SQLite3)` and all of the complexity would be hidden away.

That would be a possibility. In my opinion it is not the responsibility of find package files to set source files or even compile a library. I think this should be done outside of these files. All find package files that I encountered so far, only search for a library. I updated the commit a bit to reduce the complexity. Maybe you can agree more with that?

**Comment by nicolasfella** [comment]
> That would be a possibility. In my opinion it is not the responsibility of find package files to set source files or even compile a library. I think this should be done outside of these files. All find package files that I encountered so far, only search for a library.

That's a fair argument.

I have an idea on how to improve this further, but as-is it's a step in the right direction so I'm fine with this

**Review by nicolasfella — APPROVED** [review]

**Comment by nicolasfella** [comment]
The commit message is a bit wrong since technically macOS is Unix too

**Comment by elsiehupp** [comment]
> The commit message is a bit wrong since technically macOS is Unix too

The inverse commit message could be “Use Own SQLite3 on macOS and Windows”, since that’s what the code actually does.

**Review by allexzander — APPROVED** [review]
I like this. And I agree with the previous comment. Commit message could use some improvement :)

**Comment by FlexW** [comment]
@nicolasfella @elsiehupp @allexzander Of course you are right that the commit message is wrong. My bad. I change it.

**Pull request merged**